### PR TITLE
TIP-1287: move queries in Storage\Sql namespace

### PR DESCRIPTION
**Pull request by nmarniesse**
**Description (for Contributor and Core Developer)**

Queries was in 3 namespaces: 
`Akeneo\Pim\Enrichment\Bundle\Doctrine\ORM\Query`,  
`Akeneo\Pim\Enrichment\Bundle\Doctrine\Product\Query` and  
`Akeneo\Pim\Enrichment\Bundle\Doctrine\Storage\Sql`.  

Now queries are moved in an unique one: `Akeneo\Pim\Enrichment\Bundle\Doctrine\Storage\Sql`.  
Tests are also moved in unique namespace.  

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | -
| Added legacy Behats               | -
| Added acceptance tests            | -
| Added integration tests           | -
| Changelog updated                 | OK
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | -
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: src/Akeneo/Pim/Structure/.php_cd.php [hunk 1]
```diff
@@ -11,6 +11,7 @@
 $rules = [
     $builder->only([
         'Doctrine',
+        'Storage',
```
[the remark this answers is not in the document]
Comment (nmarniesse) [reply src/Akeneo/Pim/Structure/.php_cd.php]: Some classes in Structure use the queries that was in `Doctrine\*`. Now the queries are in `Storage\*` so I add this rule

File: CHANGELOG-4.0.md [hunk 2]
```diff
@@ -204,6 +204,29 @@
...
 - Move class from `Akeneo\Pim\Enrichment\Bundle\EventSubscriber\ComputeCompletenessOnFamilyUpdateSubscriber` to `Akeneo\Pim\Enrichment\Bundle\EventSubscriber\Family\ComputeCompletenessOnFamilyUpdateSubscriber` and mark it as final
+- Remove class `Akeneo\Pim\Enrichment\Bundle\EventSubscriber\ComputeProductModelDescendantsSubscriber`
+- Remove class `Akeneo\Pim\Enrichment\Bundle\EventSubscriber\IndexProductModelsSubscriber`
+- Update class `Akeneo\Pim\Enrichment\Component\Product\Connector\Writer\Database\MassEdit\ProductAndProductModelWriter` to remove
+    - `Akeneo\Tool\Bundle\BatchBundle\Launcher\JobLauncherInterface`,
+    - `Akeneo\Tool\Component\StorageUtils\Repository\IdentifiableObjectRepositoryInterface`,
+    - `Symfony\Component\Security\Core\Authentication\Token\Storage\TokenStorageInterface` and
+    - `string $jobName`
+- Remove class `Akeneo\Pim\Enrichment\Bundle\Doctrine\Common\Saver\ProductModelDescendantsSaver`
+- Remove class `Akeneo\Pim\Enrichment\Component\Product\Job\ComputeProductModelsDescendantsTasklet`
```
[the remark this answers is not in the document]
Comment (nmarniesse) [reply CHANGELOG-4.0.md]: Yes the lines come from this merged PR https://github.com/akeneo/pim-community-dev/pull/10835/files  

And they are removed by mistake in this PR https://github.com/akeneo/pim-community-dev/pull/10840/files

File: src/Akeneo/Pim/Enrichment/Bundle/Storage/Sql/Attribute/AttributeIsAFamilyVariantAxis.php [hunk 3]
```diff
@@ -2,7 +2,7 @@
 
 declare(strict_types=1);
 
-namespace Akeneo\Pim\Enrichment\Bundle\Doctrine\ORM\Query;
+namespace Akeneo\Pim\Enrichment\Bundle\Storage\Sql\Attribute;
```
[the remark this answers is not in the document]
Comment (nmarniesse) [reply src/Akeneo/Pim/Enrichment/Bundle/Storage/Sql/Attribute/AttributeIsAFamilyVariantAxis.php]: Yes, is it a problem? All queries here use at least a Doctrine connection. IMO this is not a reason to keep them all in `Doctrine` namespace.  

On another hand we can "easily" remove the query builder and use plain SQL. What would you do? Remove query builder in another PR or put back this classes in `Doctrine\ORM`?

[the remark this answers is not in the document]
Comment (mmetayer) [reply src/Akeneo/Pim/Enrichment/Bundle/Storage/Sql/Attribute/AttributeIsAFamilyVariantAxis.php]: leave them in `...\Storage\Sql`, we can update them later

File: src/Akeneo/Pim/Structure/.php_cd.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (nmarniesse) [reply src/Akeneo/Pim/Structure/.php_cd.php]: Fixed

File: src/Akeneo/Pim/Enrichment/Bundle/Storage/Sql/Channel/SqlGetChannelLabels.php [hunk 4]
```diff
@@ -2,7 +2,7 @@
 
 declare(strict_types=1);
 
-namespace Akeneo\Pim\Enrichment\Bundle\Product\Query\Sql;
+namespace Akeneo\Pim\Enrichment\Bundle\Storage\Sql\Channel;
```
Comment (ahocquard) [inline src/Akeneo/Pim/Enrichment/Bundle/Storage/Sql/Channel/SqlGetChannelLabels.php]: it should be moved in channel context later on.

**Pull request merged**